### ISSUE TITLE
Errors are getting lost for array elements

### ISSUE THREAD

**Issue by chess777** [yaacovCR/graphql-tools-fork#29]
I'm using graphql-tools-fork v7.2.0
NodeJS v12.8.1, tried also on v13.2.0
I have a following schema:
```
type Query {
  getType1(): Type1
}

type Type1 {
  type2: [Type2]!
}

type Type2{
  mandatoryField: String!
}
```

In Type1.type2 field resolver I return `new Error('Custom error')`.
This error gets lost, instead I receive other error `Cannot return null for non-nullable field Type2.mandatoryField.`

**Comment by yaacovCR** [comment]
Is this related to https://github.com/graphql/graphql-js/issues/2275

**Comment by yaacovCR** [comment]
To clarify, seems like the error on the array list is getting lost, not the error on the element?

**Comment by chess777** [comment]
It's not quite similar to the mentioned issue.
The one I'm encountering is: if one of the array elements fails with error, I get null in the array at that element as expected. However, instead of error returned for specific array element I get another error stating that I can't return null for non-nullable field. That is, element of type `Error` is replaced with null and error is lost.
Here is an example of resolved object:
```
{
  type2: [
    {mandatoryField: 'a'},
    {mandatoryField: 'b'},
    Error('Custom error'),
    {mandatoryField: 'd'},
  ]
}
```
And the GraphQL response:
```
{
  data: {
    getType1: {
      type2: [
        {mandatoryField: 'a'},
        {mandatoryField: 'b'},
        null,
        {mandatoryField: 'd'},
      ]      
    }
  },
  errors: [
    {
      message: 'Cannot return null for non-nullable field Type2.mandatoryField.',
      path: [
        'getType1',
        'type2',
        '2',
        'mandatoryField'
      ],
      ...
    }
  ]
}
```
Expected GraphQL response:
```
{
  data: {
    getType1: {
      type2: [
        {mandatoryField: 'a'},
        {mandatoryField: 'b'},
        null,
        {mandatoryField: 'd'},
      ]      
    }
  },
  errors: [
    {
      message: 'Custom error',
      path: [
        'getType1',
        'type2',
        '2'
      ],
      ...
    }
  ]
}
```

**Comment by yaacovCR** [comment]
What is the response from the original schema?

**Comment by yaacovCR** [comment]
Lol, I am guessing you provided it right there in the expected response...

**Comment by yaacovCR** [comment]
Looks related to https://github.com/yaacovCR/graphql-tools-fork/commit/c57c15d72c58f4d10864b6af866c275732898016

**Comment by chess777** [comment]
Yes, it's related.

**Comment by yaacovCR** [comment]
Looks to me like I should check first whether there is an associated error for the list element first and only wrap if the error is on the children.

**Comment by chess777** [comment]
Yes, you are correct

**Comment by yaacovCR** [comment]
Should be fixed by 7.2.1, but this may have caused further problems on the error proxying front as required significant refactoring, and there may be a change in edge case handling. I would love a PR to expand the test suite.
